agent/utils.mjs

```diff
@@ -0,0 +1,659 @@
+    ) {
+        if( log.verboseGet() >= log.verboseReversed().warning ) {
+            log.write( cc.warning( "Skipping automatic " ) + cc.notice( "ERC721_with_metadata" ) +
+                cc.warning( " instantiation discovery, already done before" ) + "\n" );
+        }
+        return;
+    }
+    const addressCallFrom = joAccountSC.address();
+    const addressOnSChain =
+        await waitForClonedTokenToAppear(
+            sc, "erc721_with_metadata", addressCallFrom, 40, tokensMN, strMainnetName );
+    tokenERC721SC.abi = JSON.parse( JSON.stringify( tokensMN.joABI.ERC721_with_metadata_abi ) );
+    tokenERC721SC.address = "" + addressOnSChain;
+}
+
+export async function waitForClonedTokenAppearErc1155(
+    sc, tokenERC1155SC, joAccountSC, tokensMN, strMainnetName
+) {
+    if( "abi" in tokenERC1155SC && typeof tokenERC1155SC.abi == "object" &&
+        "address" in tokenERC1155SC && typeof tokenERC1155SC.address == "string"
+    ) {
+        if( log.verboseGet() >= log.verboseReversed().warning ) {
+            log.write( cc.warning( "Skipping automatic " ) + cc.notice( "ERC1155" ) +
+                cc.warning( " instantiation discovery, already done before" ) + "\n" );
+        }
+        return;
+    }
+    const addressCallFrom = joAccountSC.address();
+    const addressOnSChain =
+        await waitForClonedTokenToAppear(
+            sc, "erc1155", addressCallFrom, 40, tokensMN, strMainnetName );
+    tokenERC1155SC.abi = JSON.parse( JSON.stringify( tokensMN.joABI.ERC1155_abi ) );
+    tokenERC1155SC.address = "" + addressOnSChain;
+}
+
+export function encodeUTF8( s ) {
+    // marshals a string to an Uint8Array,
+    // see https://gist.github.com/pascaldekloe/62546103a1576803dade9269ccf76330
+    let i = 0; const arrBytes = new Uint8Array( s.length * 4 );
+    for( let ci = 0; ci != s.length; ci++ ) {
+        let c = s.charCodeAt( ci );
+        if( c < 128 ) {
+            arrBytes[i++] = c;
+            continue;
+        }
+        if( c < 2048 )
+            arrBytes[i++] = c >> 6 | 192; else {
+            if( c > 0xd7ff && c < 0xdc00 ) {
+                if( ++ci >= s.length )
+                    throw new Error( "UTF-8 encode: incomplete surrogate pair" );
+                const c2 = s.charCodeAt( ci );
+                if( c2 < 0xdc00 || c2 > 0xdfff ) {
+                    throw new Error(
+                        "UTF-8 encode: second surrogate character 0x" +
+                        c2.toString( 16 ) + " at index " + ci + " out of range"
+                    );
+                }
+                c = 0x10000 + ( ( c & 0x03ff ) << 10 ) + ( c2 & 0x03ff );
+                arrBytes[i++] = c >> 18 | 240;
+                arrBytes[i++] = c >> 12 & 63 | 128;
+            } else
+                arrBytes[i++] = c >> 12 | 224;
+            arrBytes[i++] = c >> 6 & 63 | 128;
+        }
+        arrBytes[i++] = c & 63 | 128;
+    }
+    return arrBytes.subarray( 0, i );
+}
+
+export function decodeUTF8( arrBytes ) {
```

Comment:
Here as well https://www.npmjs.com/package/utf8

Reply:
It's not needed at all. So I will remove it.